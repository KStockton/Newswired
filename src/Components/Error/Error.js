import React from 'react';
import rott404 from '../../Assets/rottweiler-404.jpg';
import {NavLink} from 'react-router-dom'


export const Error = () => {

  return (
    <section>
      <div>
        <h3 className="Error-title">"ruff" <span role="img" aria-label="dog">&#128054;</span> 404 Page Not Found</h3>
        <NavLink className="error-back"to="/Options">Home</NavLink>
      </div>
      <img className="error" src={rott404} alt="Page Not Found"/>
    </section>
  )
};


export default Error