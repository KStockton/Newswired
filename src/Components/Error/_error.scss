.error {
  width: 100vw;
  height: 100vh;
  background-image: url('../../Assets/lost.jpg');
  background-repeat: no-repeat;
  background-position: center;
  background-size: 100% 100%;

}
.Error-title {
  display: flex;
  position: fixed;
  margin: 0 auto;
  top: 6%;
  left: 37%;
  font-size: 41px;
  color: white
}

.error-back{
  background-color: #000000c7;
  position: fixed;
  top: 16%;
  left: 46%;
  padding: 10px;
  text-decoration: none;
  color: #fff;
  border: solid 1px #186f8f;
  border-radius: 5px;
  text-transform: uppercase;
  z-index: auto;
  width: 93px;
  display: flex;
  box-shadow: 3px 3px 2px #837a7a;
  flex-direction: row;
  align-items: center;

  &:hover {
    padding-left: 10px;
    padding-right: 10px;
    padding-top: 11px;
    background-image: -webkit-gradient(linear, left bottom, left top, color-stop(0, #2cffc9), color-stop(1, #ffffff));
    color: black;
    background: #efe4e4b3;
  }
}
.error-back::before{
  background-color: #ccd0d56b;
  content: "";
  display: block;
  position: absolute;
  width: 100%;
  height: 100%;
  padding: 8px;
  left: -8px;
  top: -8px;
  border-radius: 5px;
  box-shadow: inset 0px 1px 1px #909193, 0px 1px 0px #fff;
}

.Error-fontawesome {
  font-size: 29px;
  padding-right: 8px;
}

 
