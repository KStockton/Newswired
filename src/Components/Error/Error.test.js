import React from 'react';
import Error from './Error';
import { NavLink } from 'react-router-dom';
import { shallow } from 'enzyme';

describe('Error', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Error/>) 
  })
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  });

  it('Navlink should have a prop of /Options with a link if something goes wrong', () => {
     expect(wrapper.find(NavLink).props().to).toBe('/Options');
   });
});