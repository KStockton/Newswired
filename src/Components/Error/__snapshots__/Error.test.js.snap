// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error should match the snapshot 1`] = `
<section>
  <h3
    className="Error-title"
  >
    Error 404 Page Not Found
  </h3>
  <img
    alt="Page Not Found"
    className="error"
    src="rottweiler-404.jpg"
  />
</section>
`;
